File: Heat/ModelBuilders/PlantServiceModelViewBuilder.vb
Imports Heat.ViewModels.PlantServices

Public Class PlantServiceModelViewBuilder

    Private _db As IHeatDBContext

    Sub New(dbContext As IHeatDBContext)
        _db = dbContext
    End Sub

    Public Function GetCreatePlantServiceViewModel(plantID As Integer) As CreatePlantServiceViewModel
        Dim result As New CreatePlantServiceViewModel

        result.PlantID = plantID
        result.PlantDescription = _db.Plants.Find(plantID).Name

        Return result

    End Function

End Class

File: Heat.ConvertedToC#/ModelBuilders/PlantServiceModelViewBuilder.cs
using Heat.ViewModels.PlantServices;
namespace Heat
{

    public class PlantServiceModelViewBuilder
	{


		private IHeatDBContext _db;
		public PlantServiceModelViewBuilder(IHeatDBContext dbContext)
		{
			_db = dbContext;
		}

		public CreatePlantServiceViewModel GetCreatePlantServiceViewModel(int plantID)
		{
			CreatePlantServiceViewModel result = new CreatePlantServiceViewModel();

			result.PlantID = plantID;
			result.PlantDescription = _db.Plants.Find(plantID).Name;

			return result;

		}

	}
}
